file-formats/saia/type/zif_aff_saia_v1.intf.abap

```diff
@@ -0,0 +1,142 @@
+
+  CONSTANTS:
+    "! <p class="shorttext">Number of Focuses Resources</p>
+    "! Number of focused resources
+    BEGIN OF co_number_of_focused_resources,
+      "! <p class="shorttext">Exactly One</p>
+      "! Exactly one
+      exactly_one   TYPE ty_number_of_focused_resources VALUE 'EXACTLY_ONE',
+      "! <p class="shorttext">At Least One</p>
+      "! At least one
+      at_least_one  TYPE ty_number_of_focused_resources VALUE 'AT_LEAST_ONE',
+      "! <p class="shorttext">More Than One</p>
+      "! More than one
+      more_than_one TYPE ty_number_of_focused_resources VALUE 'MORE_THAN_ONE',
+    END OF co_number_of_focused_resources.
+
+  TYPES:
+    "! <p class="shorttext">Action Filter</p>
+    "! Filtering of action according to specified applicable criteria.
+    "! If filter is satisfied then the action is available and will be shown.
+    "! $required
+    BEGIN OF ty_filters,
+      "! <p class="shorttext">Number of Focused Resources</p>
+      "! Number of Focused Resources
+      number_of_focused_resources TYPE ty_number_of_focused_resources,
+      "! <p class="shorttext">Supported Development Object Types</p>
+      "! Supported development object types
+      supported_dev_object_types  TYPE ty_development_object_types,
+      "! <p class="shorttext">Supported Resource Types</p>
+      "! Supported resource types
+      supported_resource_types    TYPE ty_resource_types,
+    END OF ty_filters.
+
+  "! <p class="shorttext">ABAP Class Name</p>
+  "! ABAP class name
+  TYPES ty_class_name TYPE c LENGTH 30.
+
+  TYPES:
+    "! <p class="shorttext">ADT IDE Action</p>
+    "! ADT IDE Action
+    "! $required
+    BEGIN OF ty_adt_saia_object,
+      "! <p class="shorttext">Unique Id of the Action</p>
+      "! Unique action id - it should be human readable and reveal the intention of the action
+      "! Action id is case insensitive
+      "! TODO: length restriction, check how long it should be, su22 variant restrictions???
+      "! $required
+      action_id                      TYPE string,
+      "! <p class="shorttext">Action Title</p>
+      "! Action title for the action
+      "! $required
+      title                          TYPE string,
+      "! <p class="shorttext">Action Description</p>
+      "! What is the action doing and how can it be used
+      "! $required
+      description                    TYPE string,
+      "! <p class="shorttext">Action Filter(In development)</p>
+      "! Filter for application of action according to specific object type or object type groups
+      "! TODO: new filter mechanis currently under development -> all filters stored in a string on database
+      "! $required
+      filters                        TYPE ty_filters,
+      "! <p class="shorttext">Action Handler Class</p>
+      "! Implementation class for handling the action input. Needs to implement interface  {@link if_aia_action }.
+      "! $required
+      implementation_class           TYPE ty_class_name,
```

Comment:
Again, our recommendation would be to keep title and the name of the component in sync.